src/Components/Patient/PatientPrivacyToggle.tsx

```diff
@@ -0,0 +1,153 @@
+import * as Notification from "../../Utils/Notifications.js";
+import CareIcon from "../../CAREUI/icons/CareIcon.js";
+import { ConsultationModel } from "../Facility/models.js";
+import { useDispatch } from "react-redux";
+import { useEffect, useState } from "react";
+import { UserRole } from "../../Common/constants.js";
+import {
+  getConsultationBed,
+  togglePatientPrivacy,
+} from "../../Redux/actions.js";
+import useAuthUser from "../../Common/hooks/useAuthUser.js";
+interface PatientPrivacyToggleProps {
+  consultationId: string;
+  consultation?: ConsultationModel | null;
+  fetchPatientData?: (state: { aborted: boolean }) => void;
+}
+export default function PatientPrivacyToggle(props: PatientPrivacyToggleProps) {
+  const { consultationId, consultation, fetchPatientData } = props;
+  const [privacy, setPrivacy] = useState<boolean>(false);
+  const user = useAuthUser();
+  const dispatch: any = useDispatch();
+
+  //condititonally render the privacy toggle button depending on user role
+  const allowPrivacyToggle = () => {
+    const currentUserType: UserRole = user.user_type;
+    if (
+      currentUserType == "DistrictAdmin" ||
+      currentUserType == "StateAdmin" ||
+      currentUserType == "LocalBodyAdmin" ||
+      (currentUserType == "Doctor" &&
+        user?.home_facility_object?.id === consultation?.facility) ||
+      (currentUserType == "Staff" &&
+        user?.home_facility_object?.id === consultation?.facility) ||
+      currentUserType == "WardAdmin"
+    )
+      return true;
+
+    return false;
+  };
+  //hook to fetch the privacy info of the patient
+  useEffect(() => {
+    const getPrivacyInfo = async () => {
+      if (
+        consultation?.current_bed?.privacy == true ||
+        consultation?.current_bed?.privacy == false
```

Comment:
what are we checking here, why are you doing both true and false?

Reply:
I guess he is trying to check if the value exists for that variable